packages/optimizer-express/lib/AmpOptimizeMiddleware.js

```diff
@@ -133,16 +132,16 @@ class AmpOptimizerMiddleware {
         const ampOptimizerParams = req.ampOptimizerParams || {};
         ampOptimizerParams.ampUrl = linkRelAmpHtmlUrl;
         ampOptimizerParams.ampRuntimeVersion = version;
-        try {
-          let body = Buffer.concat(chunks).toString('utf8');
-          if (!(options.ampOnly && !isAmp(body))) {
+        let body = Buffer.concat(chunks).toString('utf8');
+        if (!(options.ampOnly && !isAmp(body))) {
+          try {
             body = await optimizer.transformHtml(body, ampOptimizerParams);
+          } catch (err) {
+            console.error('Error applying AMP Optimizer. Sending original page', err);
           }
-          res.setHeader('Content-Length', Buffer.byteLength(body, 'utf-8'));
-          res.end(body, 'utf-8');
-        } catch (err) {
-          console.error('Error applying AMP Optimizer. Sending original page', err);
```

Comment:
@andreban Did this work? If yes, why?

Reply:
It did work -> It used to call `res.end()` inside a catch block before it was migrated to async / await in this change: https://github.com/ampproject/amp-toolbox/commit/17856f6133d3662dc9684c32391f47e49bcbeba0#diff-c8065ddd90540be59b6bd37efc257e38

Reply:
Hah!